### PR TITLE
4741 filtering on aggregates does not work

### LINKED ISSUE #4741: Filtering on Aggregates does not work
**Describe the bug**
Feature that was supposed to be added in #4308 does not work.

**To Reproduce**
Add count_GT on NodeConnection in where

**Expected behavior**
Should filter on aggregates

**Example**

Please note that output Cypher query never mentions any value close to `count_GT`. In the output I receive (:Oli) nodes with totalCount equal to 1, which is invalid due to `count_GT = 1`

Query:
```graphql
query Node($first: Int, $where: OpportunityWhere) {
  opportunitiesConnection(first: $first, where: $where) {
    edges {
      node {
        country
        currency
        department
        listsOlisConnection {
          totalCount
        }
      }
    }
  }
}
```

Params:
```json
{
  "first": 10,
  "where": {
    "listsOlisAggregate": {
      "count_GT": 1
    }
  }
}
```

Cypher query:
```cypher
MATCH (this0:Opportunity)
WHERE ($isAuthenticated = true AND EXISTS {
    MATCH (this0)<-[this1:CAN_SEE]-(this2:User)
    WHERE ($jwt.sub IS NOT NULL AND this2.id = $jwt.sub)
})
WITH collect({ node: this0 }) AS edges
WITH edges, size(edges) AS totalCount
CALL {
    WITH edges
    UNWIND edges AS edge
    WITH edge.node AS this0
    WITH *
    
    LIMIT $param2
    CALL {
        WITH this0
        MATCH (this0)-[this3:LISTS]->(this4:Oli)
        WHERE ($isAuthenticated = true AND EXISTS {
            MATCH (this4)<-[this5:LISTS]-(this6:Opportunity)
            WHERE EXISTS {
                MATCH (this6)<-[this7:CAN_SEE]-(this8:User)
                WHERE ($jwt.sub IS NOT NULL AND this8.id = $jwt.sub)
            }
        })
        WITH collect({ node: this4, relationship: this3 }) AS edges
        WITH edges, size(edges) AS totalCount
        CALL {
            WITH edges
            UNWIND edges AS edge
            WITH edge.node AS this4, edge.relationship AS this3
            RETURN collect({ node: { __resolveType: "Oli", __id: id(this4) } }) AS var9
        }
        RETURN { edges: var9, totalCount: totalCount } AS var10
    }
    RETURN collect({ node: { country: this0.country, currency: this0.currency, department: this0.department, listsOlisConnection: var10 } }) AS var11
}
RETURN { edges: var11, totalCount: totalCount } AS this {"isAuthenticated":true,"jwt":{"sub":<truncated>,"email":<truncated>},"param2":{"low":10,"high":0}} 
```

https://discord.com/channels/787399249741479977/1207323181841514656

### PR DESCRIPTION
**Pull request by angrykoala**
# Description

Fix filtering on aggregation fields in connections

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Thanks for the Neo4j GraphQL Toolbox updates.

The Neo4j GraphQL Toolbox has now been torn down - reopening this PR will republish it.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3404 | 3404 | 39 | 43 | 1134 |
|  aggregations.TopLevelAggregateWithMultipleFields | 6802 | 6802 | 49 | 52 | 1134 |
|  aggregations.NestedAggregation | 16544 | 16544 | 58 | 70 | 2174 |
|  aggregations.AggregationWithWhere | 11970 | 11970 | 36 | 61 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116478 | 22116478 | 2214 | 2315 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116478 | 22116478 | 2181 | 2240 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9701 | 9701 | 36 | 48 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9928 | 9928 | 32 | 38 | 2174 |
|  aggregations.DeeplyNestedCount | 11062184 | 11062184 | 2400 | 2517 | 2008534 |
|  aggregations.InterfacesAggregations | 6242 | 6242 | 40 | 49 | 2080 |
|  aggregations.InterfacesAggregationsWithTwoFields | 11444 | 11444 | 72 | 93 | 2080 |
|  batch-create.BatchCreate | 4200 | 4200 | 116 | 181 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 47 | 54 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 140 | 191 | 3003 |
|  connections.Connection | 14049 | 14049 | 87 | 65 | 2174 |
|  connections.NestedConnection | 41068 | 41068 | 119 | 152 | 4516 |
|  create.SimpleMutation | 7 | 7 | 47 | 58 | 1 |
|  cypher-directive.TopLevelMutationDirective | 1135 | 1135 | 25 | 25 | 1134 |
|  delete.SimpleDelete | 19401 | 19401 | 191 | 200 | 1040 |
|  delete.NestedDeleteInUpdate | 19165 | 19165 | 121 | 201 | 2040 |
|  2871.NestedRelationshipFilter | 23083 | 23083 | 51 | 58 | 4431 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 37 | 42 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22891 | 22891 | 66 | 85 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 33 | 38 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 48 | 64 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 17 | 25 | 1040 |
|  query.SimpleQueryWithRelationship | 16129 | 16129 | 35 | 40 | 2174 |
|  query.QueryWhere | 9676 | 9676 | 29 | 52 | 2161 |
|  query.SimpleQueryWithNestedWhere | 9843 | 9843 | 44 | 56 | 2161 |
|  query.Nested | 10095502 | 10095502 | 6921 | 6832 | 2008534 |
|  query.NestedWithFilter | 10074351 | 10074351 | 6751 | 6925 | 2004000 |
|  query.OrFilterOnRelationships | 42853 | 42618 | 136 | 169 | 1763 |
|  query.OrFilterOnRelationshipsAndNested | 36112 | 36079 | 172 | 217 | 1763 |
|  query.QueryWithNestedIn | 13868 | 13451 | 47 | 62 | 1383 |
|  query.NestedConnectionWhere | 9801 | 9801 | 46 | 55 | 2174 |
|  query.DeeplyNestedConnectionWhere | 9859 | 9859 | 74 | 97 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 18758 | 18758 | 114 | 155 | 1586 |
|  query.NestedWithRelationshipSingleFilters | 3865 | 3865 | 143 | 181 | 1134 |
|  query.Fulltext | 64 | 64 | 27 | 31 | 16 |
|  query.FulltextWithNestedQuery | 558 | 558 | 37 | 41 | 84 |
|  sorting-and-cypher.TopLevelSortWithCypher | 14059 | 14059 | 39 | 51 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypher | 14059 | 14059 | 56 | 63 | 2174 |
|  sorting-and-cypher.TopLevelSortWithCypherWithNested | 14204 | 14204 | 50 | 56 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypherWithNested | 14204 | 14204 | 94 | 98 | 2174 |
|  sorting-and-cypher.TopLevelSortWithExpensiveCypher | 14933 | 14933 | 94 | 106 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithExpensiveCypher | 14404 | 14404 | 93 | 102 | 2174 |
|  sorting.SortMultipleTypes | 3498 | 3498 | 77 | 98 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 14435 | 14435 | 97 | 114 | 2174 |
|  sorting.SortOnNestedFields | 14049 | 14049 | 39 | 51 | 2174 |
|  sorting.SortDeeplyNestedFields | 43148 | 43148 | 74 | 112 | 4516 |
|  sorting.ConnectionWithSort | 3281 | 3281 | 76 | 76 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 51 | 57 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 49 | 62 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 231 | 306 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 221 | 244 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 95 | 123 | 2002 |

</details>
        
Old Schema Generation: 25.597s
Schema Generation: 25.504s
Old Subgraph Schema Generation: 28.933s
Subgraph Schema Generation: 29.205s

**Review by darrellwarde — DISMISSED** [review]
LGTM!

File: .changeset/wet-onions-crash.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@neo4j/graphql": patch
+---
+
+Fix filtering on aggregation fields in connections
```
Comment (MacondoExpress) [inline .changeset/wet-onions-crash.md]:
```suggestion
Fix aggregation filters for connection fields
```

**Review by MacondoExpress — DISMISSED** [review]

**Pull request merged**